csrc/python_frontend/python_bindings.cpp

```diff
@@ -271,29 +271,43 @@ std::vector<std::optional<bool>> computeContiguity(
 // `stride order` vector corresponds to the order for each logical domain in
 //     physical memory; For any 0 <= i < n , we know the dimension i has the
 //     stride_order[i]-th smallest stride.
+//     An exception to this are implicit broadcast dimensions, i.e. dimensions
+//     with `stride == 0`, where we would maintain their semantical position
 // `contiguity` vector to whether or not indexing could be collaped
 //     corresponding to each physical domain;
 //
 // e.g. Given size and stride as follow:
-//   sizes   = [2, 1, 3, 1, 4, 3]
-//   strides = [12, 4, 4, 4, 1, 0]
-// we would compute stride order as: [5, 4, 3, 2, 1, 0]. Since the original
-// stride is in descending order. Note that there's more than one way to define
-// a stride order when we have equal strides. In the context of index
-// collapsing, how we resolve that shouldn't matter, hence we just go with
-// preserving their original order. Similarly, we compute contiguity as: [True,
-// None, True, None, True, None], Since the physical order is the same as the
-// logical order, this one is trivial to compute.
+//   sizes   = [2, 2, 2, 2]
+//   strides = [8, 4, 2, 1]
+// Obviously the stride order as: [3, 2, 1, 0] for row-major order, i.e. stride
+// in descending order and contiguity flag will be [True, True, True, True]
 //
 // e.g. Given size and stride as follow:
-//   sizes   = [2, 3, 1, 5, 4]
-//   strides = [28, 4, 14, 0, 1]
-// stride_order would be: [4, 2, 3, 0, 1], marking the order of strides in the
-// vector. Meanwhile, contiguity would be computed on the physical domain, i.e.
-// on sorted sizes & strides.
-//   sorted_size    = [2, 1, 3, 4, 5]
-//   sorted_strides = [28, 14, 4, 1, 0]
-//   contiguity would be: [False, None, True, True, None]
+//   sizes   = [2, 1, 3, 1, 4]
+//   strides = [24, 4, 8, 4, 2]
+// Note that there are a few explicit broadcast dimensions, dimensions with size
+// == 1. The stride for explicit broadcast dimensions participates in stride
```

Comment:
Is <size=1, stride=0> considered implicit or explicit?

Reply:
Good catch!

The documentation here says it's explicit, but the implementation is treating it as implicit, since it only checks for `stride == 0`.
Let me clean it up and add a test for that.